kubernetes/client_factory.go

```diff
@@ -210,26 +240,53 @@ func (cf *clientFactory) newSAClient(clusterInfo *RemoteClusterInfo) (*K8SClient
 	}
 }
 
+// getClient returns a client for the specified token. Creating one if necessary.
+func (cf *clientFactory) GetSAClients() map[string]ClientInterface {
+	return cf.saClientEntries
```

Comment:
This does not look safe. This is using the saClientEntries map without the mutex. Everywhere else in here "saClientEntries" access is protected around the mutex lock. I suspect this function will need to use the mutex also.